cmake/onnxruntime.cmake

```diff
@@ -282,44 +282,77 @@ endif()
 
 # Assemble the Apple static framework (iOS and macOS)
 if(onnxruntime_BUILD_APPLE_FRAMEWORK)
+  if(${CMAKE_SYSTEM_NAME} STREQUAL "iOS")
+    set(STATIC_FRAMEWORK_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}-${CMAKE_OSX_SYSROOT})
+  else() # macOS
+    set(STATIC_FRAMEWORK_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR})
+  endif()
+
+  # Setup the various directories required. Remove any existing ones so we start with a clean directory.
```

Comment:
should we move the directory setup to post build commands? it may be easier to ensure that we get a clean output directory for each build there.
the original version also did this at build generation time but I'm not sure that's the best place for it.

Reply:
Added as pre_build for onnxruntime target.